### ISSUE TITLE
RFE: Add a shortcut to use a Koji side tag

### ISSUE THREAD

**Issue by hroncok** [rpm-software-management/mock#1295]
Mock already has an option to enable the Koji repository with `--enablerepo=local`.
It would be really convenient to have a similar option to enable a Koji side-tag repo.

Currently, this is possible with something like:

```
mock -r centos-stream-9-x86_64 --addrepo='https://kojihub.stream.centos.org/kojifiles/repos/c9s-build-side-2685-stack-gate/latest/x86_64/'
mock -r fedora-rawhide-x86_64 --addrepo='https://kojipkgs.fedoraproject.org/repos/f40-build-side-81735/latest/x86_64/'
```

However, that's rather long and hard to remember.

I would appreciate it if I could do something like `--side-tag=f40-build-side-81735`.

I see several options to achieve this.

### Add side_tag variable to the local repo baseurl

E.g. replace this:

```
[local]
name=local
baseurl=https://kojipkgs.fedoraproject.org/repos/rawhide/latest/$basearch/
...
enabled={{ not mirrored }}
...
```

With:

```
[local]
name=local
baseurl=https://kojipkgs.fedoraproject.org/repos/{{ side_tag or "rawhide" }}/latest/$basearch/
...
enabled={{ side_tag != None or not mirrored }}
...
```

The `side_tag` would obviously be `None` by default and as an user I could override it by `--side-tag=...`

Note: This could be challenging with the combined centos-stream+epel repos, where there are 2 local repositories, one for each Koji, but I guess not unsolvable.

### Add common --addrepo aliases

This option is more general but a tad more flexible.

In short, assume I could do: `--addrepo=koji://f40-build-side-81735` or `--addrepo=copr://@python/python3.13`.

Such aliases would need to be well-defined and doing it this way requires more thought. Copr itself already has the concept for `copr://` when adding external repositories, so this could be reused in Copr and more users could benefit from this.

**Comment by praiskup** [comment]
Wouldn't a simple script `koji-side-tag-mock-config` generate the mock-config (with include the default) help here?

**Comment by praiskup** [comment]
Otherwise, good idea with the `--addrepo`.  If anyone wanted to submit patch, here goes the code that needs to be updated:

https://github.com/rpm-software-management/mock/blob/4b3fb22e2f74a17bf7879395b6f6418192430163/mock/py/mock.py#L773-L775

The `copr://` syntax is already implemented in copr-frontend:
https://github.com/fedora-copr/copr/blob/3b68ad13a4f44229ecb02ef5d9677278a7a69c2e/frontend/coprs_frontend/coprs/helpers.py#L607-L630

Note that `copr://` means something different for Fedora, for openEuler, and for Red Hat Copr users. So it should be configurable. Ditto Koji.

**Comment by Conan-Kudo** [comment]
Probably `copr://<baseurl>/<namespace>/<project>/<root>`, `koji://<baseurl>/<tag>/`, and `obs://<baseurl>/<project>/<repository>` URIs would be good to have here.

**Comment by hroncok** [comment]
> Wouldn't a simple script `koji-side-tag-mock-config` generating the mock-config (with include the default) help here?

Yes, but that would help me, not others.
